Mathlib/Probability/CondJensen.lean

```diff
@@ -0,0 +1,577 @@
+/-
+Copyright (c) 2025 Yongxi Lin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yongxi Lin, Thomas Zhu
+-/
+import Mathlib.Algebra.Order.Ring.Star
+import Mathlib.Data.Int.Star
+import Mathlib.Data.Real.StarOrdered
+import Mathlib.GroupTheory.MonoidLocalization.Basic
+import Mathlib.MeasureTheory.Function.ConditionalExpectation.Basic
+import Mathlib.Order.CompletePartialOrder
+import Mathlib.Topology.Algebra.Module.ModuleTopology
+import Mathlib.Topology.Compactness.PseudometrizableLindelof
+
+/-!
+# Conditional Jensen's Inequality
+
+This file contains the proof of the conditional Jensen's inequality.
+
+## Main Statement
+
+* `conditional_jensen`: the conditional Jensen's inequality.
+
+## References
+
+* [Hytönen, Tuomas, Jan Van Neerven, Mark Veraar, and Lutz Weis. Analysis in Banach spaces.
+  Springer, 2016.][Hytonen_VanNeerven_Veraar_Wies_2016]
+-/
+
+open MeasureTheory ProbabilityTheory TopologicalSpace Set Metric ContinuousLinearMap RCLike
+open scoped ENNReal
+
+namespace RCLike
+
+/-- Lemma 1.2.9 in [Hytonen_VanNeerven_Veraar_Wies_2016]: a closed convex set is the intersection of
+  countably many half spaces in a separable space. -/
+theorem iInter_halfSpaces_eq_of_separableSpace {𝕜 E : Type*} {s : Set E}
+    [RCLike 𝕜] [NormedAddCommGroup E] [NormedSpace ℝ E]
+    [SeparableSpace E] [Module 𝕜 E] [ContinuousSMul 𝕜 E]
+    (hs₁ : Convex ℝ s) (hs₂ : IsClosed s) (hs₃ : s.Nonempty) :
+    ∃ (L : ℕ → E →L[𝕜] 𝕜) (c : ℕ → ℝ),
+    ⋂ (i : ℕ), {x | re ((L i) x) - c i ≥ 0} = s ∧
```

Comment:
In Mathlib, for uniformity and to avoid duplication of some lemmas, we state all inequalities with `≤`, not `≥`. Can you change that everywhere in your code?
```suggestion
    ⋂ (i : ℕ), {x | 0 ≤ re ((L i) x) - c i} = s ∧
```
Also, why not `{x | c i ≤ re ((L i) x)}`?